### PR TITLE
[PLATFORM-1142] Meatball menus for stream editor

### PR DESCRIPTION
**Pull request by tumppi**
Moved some actions (streamId copy, field delete) behind a meatball menu on input.

Refactored `TextField` to support providing actions array, which contains `DropdownActions.Item`s.

### CODE REVIEW

File: app/src/userpages/components/StreamPage/Show/InfoView/index.jsx [hunk 1]
```diff
@@ -24,6 +23,7 @@ type Props = {
 export const InfoView = ({ disabled }: Props) => {
     const stream = useSelector(selectEditedStream)
     const dispatch = useDispatch()
+    /* eslint-disable-next-line no-unused-vars */
```
Comment (juhah) [inline app/src/userpages/components/StreamPage/Show/InfoView/index.jsx]: Or just leave out `isCopied`? -> `const { copy } = useCopy()`

File: app/stories/shared.stories.jsx [hunk 2]
```diff
@@ -235,6 +235,24 @@ story('Text Field/Text')
         <TextInput preserveLabelSpace label="With invalid value" value="Something invalid" error="Oh, something went wrong!" />
     ))
 
+story('Text Field/With actions')
+    .addWithJSX('basic', () => (
```
Comment (juhah) [inline app/stories/shared.stories.jsx]: Might be good a disabled state here as well?

File: app/src/userpages/components/StreamPage/Show/InfoView/index.jsx (hunk 1 above)
Comment (tumppi) [reply app/src/userpages/components/StreamPage/Show/InfoView/index.jsx]: Oh, I thought it was returning an array 🤦‍♂

**Comment by tumppi** [comment]
Good points Juha, fixed all but the z-index ones. @mondoreale: Any idea why that weird stacking is happening and how to fix it?

**Comment by mondoreale** [comment]
@tumppi there are 2 fixes I see.

1. Give later sortable items lower zIndex. It won't fix the issue if dropdowns render upwards though. Patch:
[fix-overlap.txt](https://github.com/streamr-dev/streamr-platform/files/3926859/fix-overlap.txt)

2. Tell `shared_textField_actionContainer` of the currently open dropdown that its z-index should be higher than anyone else's.

It's always the same story when it comes down to lists with relatively positioned items…

File: app/src/userpages/components/KeyField/index.jsx [hunk 3]
```diff
@@ -145,6 +144,8 @@ class KeyField extends React.Component<Props, State> {
         }
     }
 
+    insertIf = (condition: boolean, ...elements: Array<any>) => (condition ? elements : [])
```
[the remark this answers is not in the document]
Comment (mondoreale) [reply app/src/userpages/components/KeyField/index.jsx]: Also the name doesn't really fit its purpose. I don't have a better one tho, lol. Maybe `useIf(value: any, defaultValue: any, condition: boolean) => codition ? value : defaultValue`. I don't know.  Maybe we don't even need it if it's just another `:?`. Naming is hard.

[the remark this answers is not in the document]
Comment (mondoreale) [reply app/src/userpages/components/KeyField/index.jsx]: Oh, and `...elements` doesn't simply rewrites `elements` if the condition is met. It wrapps it in another array, i.e.
```js
insertIf(true, [1]) //=> [[1]]
```
Is that intended?

[the remark this answers is not in the document]
Comment (tumppi) [reply app/src/userpages/components/KeyField/index.jsx]: Not intended indeed. Good spot! Noticed the same while fixing test cases.

**Comment by tumppi** [comment]
Thanks for the comments guys! Everything should be fixed now.

**Comment by mondoreale** [comment]
The `z-index` fix u applied fixes the main issue. There's the other one I mentioned, but maybe we can just leave it out:

Screenshot 2019-12-11 13 44 42

Looks like being "sortable" causes it. Access keys don't share that behaviour.

Screenshot 2019-12-11 13 47 56

**Comment by tumppi** [comment]
@mondoreale: leaving that out for now sounds good. It's quite a marginal case anyway.

**Pull request merged**